sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2914,15 +2914,17 @@ lsc_atomic_update(T *p, Toffset offset, __ESIMD_NS::simd<T, N> src0,
 template <__ESIMD_NS::atomic_op Op, typename T, int N,
           lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
-          typename AccessorTy>
+          typename AccessorTy, typename Toffset>
 __ESIMD_API std::enable_if_t<!std::is_pointer<AccessorTy>::value,
                              __ESIMD_NS::simd<T, N>>
-lsc_atomic_update(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
+lsc_atomic_update(AccessorTy acc, __ESIMD_NS::simd<Toffset, N> offsets,
                   __ESIMD_NS::simd_mask<N> pred) {
 #ifdef __ESIMD_FORCE_STATELESS_MEM
   return lsc_atomic_update<Op, T, N, DS, L1H, L3H>(
       __ESIMD_DNS::accessorToPointer<T>(acc), offsets, pred);
 #else
+  static_assert(std::is_integral_v<Toffset> && sizeof(Toffset) == 4,
```

Comment:
Generally speaking it is backward incompatibility. It was possible to pass simd<uint64_t,N> Offsets and have automatic conversion at the call-site to simd<uint32_t, N> Offsets.
It may be Ok here taking into account 1) The changes are in experimental 2) It is pretty much a corner case 3) It is a time-windows for api/abi breaking changes now.
@turinevgeny , @sarnex - what do you think?

Reply:
Thank you. I also created a test and reproduced such fail:
```
/include/sycl/ext/intel/experimental/esimd/memory.hpp:2920:1: note: candidate template ignored: could not match 'unsigned int' against 'unsigned long'                                                                                                             
lsc_atomic_update(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
```
It is good then.